2015-05-03  Jingtao Wang  <classong@lnxsrv02.seas.ucla.edu>

        * timespec.h (timespec_cmp): set return value as 1 and -1 for positive
	        and negative diff respectively, return a.tv_nsec - b.tvdec when diff is
		        zero.
--- timespec.h	2015-05-02 23:54:54.864704000 -0700
+++ lab5.h	2015-05-03 00:22:52.254616000 -0700
@@ -46,7 +46,12 @@
 timespec_cmp (struct timespec a, struct timespec b)
 {
   int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (diff > 0)
+    return 1;
+  else if (diff<0)
+    return -1;
+  else
+    return a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
